### DISCUSSION TITLE
shell login to container running with other user and systemd

### DISCUSSION THREAD

**Discussion by JuSchi89** [containers/podman#23340] [Q&A] [answered]
hey guys,

sorry for a kinda newbie question but I am new in the container universe and wonder how I manage to login to a shell of a running container. Each container has an own user which is also set in the systemd service. The containers work and all run with unpriviliged users but I can't manage to log into their shells.

I have an account with sudo rights where I do the management and set up podman (uid=1000). From there I was able to do a podman ps of other containers with something like:

`systemd-run --uid=1001 --pty --wait --collect --service-type=exec /usr/bin/podman ps -a`

Like that I can also see the container ID. But how am I able to connect to the shell? I adjusted the command `podman exec -it <Container ID> /bin/bash` in different ways to fit the systemd-run command, but I always get `Error: crun: error opening file /run/user/1001/crun/3723bb2561662cf19d78c43eb4825e269d870315231b06f3ed7532035516aba7/status: No such file or directory: OCI runtime attempted to invoke a command that was not found`

Then I also tried to switch to the user which runs the container to enter the shell from there but then I get:
`ERRO[0000] XDG_RUNTIME_DIR directory "/run/user/1000" is not owned by the current user`

What am I doing wrong?

**Comment by rhatdan** [comment, marked as answer]
You need a full login session to run podman.  Try:

` #  machinectl shell -q USERNAME@ /usr/bin/podman exec ...`

Although I don't really understand your goal.

**Comment by rhatdan** [comment]
You might be looking for podmansh as well

$ man podmansh
